### PR TITLE
Update FreeBSD NFS synced folder to support mount_options

### PR DESCRIPTION
**Pull request by amontalban**
Fix for #7466.

PS: I do not have any Ruby experience, I just copied Linux capability and modified for FreeBSD. Couldn't test it in FreeBSD because I don't know how to build an installer to test it from scratch.

### CODE REVIEW

File: plugins/guests/freebsd/cap/mount_nfs_folder.rb [hunk 1]
```diff
@@ -1,22 +1,41 @@
...
+
+            # Mount
+            hostpath = opts[:hostpath].dup
+            hostpath.gsub!("'", "'\\\\''")
+
+            # Figure out any options
+            mount_opts = ["nfsv#{opts[:nfs_version]}"]
+            mount_opts << "udp" if opts[:nfs_udp]
+            if opts[:mount_options]
+              mount_opts = opts[:mount_options].dup
```
Comment (sethvargo) [inline plugins/guests/freebsd/cap/mount_nfs_folder.rb]: This is going to overwrite the version.

Comment (amontalban) [inline plugins/guests/freebsd/cap/mount_nfs_folder.rb]: Correct, I just copied https://github.com/mitchellh/vagrant/blob/master/plugins/guests/linux/cap/mount_nfs.rb and modified some things.

Want me to submit a fix for Linux too?

File: plugins/guests/freebsd/cap/mount_nfs_folder.rb [hunk 2]
```diff
@@ -1,22 +1,41 @@
...
+            # Expand the guest path so we can handle things like "~/vagrant"
+            expanded_guest_path = machine.guest.capability(
+              :shell_expand_guest_path, opts[:guestpath])
+
+            # Do the actual creating and mounting
+            commands << "mkdir -p '#{expanded_guest_path}'"
+
+            # Mount
+            hostpath = opts[:hostpath].dup
+            hostpath.gsub!("'", "'\\\\''")
```
Comment (sethvargo) [inline plugins/guests/freebsd/cap/mount_nfs_folder.rb]: Can you explain what this is doing?

Comment (sethvargo) [inline plugins/guests/freebsd/cap/mount_nfs_folder.rb]: I mean - I know what it's doing, but can you explain _why_?

**Pull request closed without merging**